File: front_dev/src/domain/admin/qa.ts
// QA管理の型定義（IMPL-202608060837 T16 / Knowledge MCP の QaSummary/QaDetail に対応）

export interface QaSummary {
  id: string;
  title: string;
  category: string | null;
  tags: string[];
  hiroba_question_altered_count: number;
}

export interface QaTagRef {
  id: number;
  name: string;
}

export interface QaCategoryRef {
  id: number;
  name: string;
}

export interface QaDetail {
  id: string;
  title: string;
  question_text: string;
  answer_text: string;
  category: QaCategoryRef | null;
  tags: QaTagRef[];
  hiroba_question_altered_count: number;
}

export interface QaListResponse {
  items: QaSummary[];
  total: number;
}

export interface QaListParams {
  keyword?: string;
  category?: string;
  tag_id?: number[];
  limit?: number;
  offset?: number;
}

export interface QaCreateRequest {
  title: string;
  question_text: string;
  answer_text: string;
  category_id?: number | null;
  tag_ids?: number[];
}

// 未指定フィールドは「変更なし」。tag_ids は [] で全解除。
export interface QaUpdateRequest {
  title?: string;
  question_text?: string;
  answer_text?: string;
  category_id?: number | null;
  tag_ids?: number[];
}

export interface Category {
  id: number;
  name: string;
}

// CSV 一括インポート結果（IMPL-202608261022 T11/T12）。
export interface QaImportRowResult {
  row: number;
  status: "success" | "error";
  qa_id?: string | null;
  error?: string | null;
}

export interface QaImportResponse {
  total: number;
  success_count: number;
  results: QaImportRowResult[];
}

File: knowledge_mcp/src/knowledge_mcp/models/qa.py
"""QA管理用データモデル（実装指示書 4.1 / T1）。

QaSummary（一覧行）/ QaDetail（詳細）と、QA登録編集の業務エラー QaError を定義する。
"""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import List, Optional


class QaError(Exception):
    """QA登録編集の業務エラー（必須項目欠落、存在しないcategory_id/tag_ids指定、対象QA不存在等）。"""


@dataclass
class QaSummary:
    id: str
    title: str
    category: Optional[str]
    tags: List[str] = field(default_factory=list)
    question_altered_count: int = 0

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "title": self.title,
            "category": self.category,
            "tags": list(self.tags),
            "question_altered_count": self.question_altered_count,
        }


@dataclass
class QaDetail:
    id: str
    title: str
    question_text: str
    answer_text: str
    category: Optional[dict] = None      # {"id": int, "name": str}
    tags: List[dict] = field(default_factory=list)  # [{"id": int, "name": str}, ...]
    question_altered_count: int = 0

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "title": self.title,
            "question_text": self.question_text,
            "answer_text": self.answer_text,
            "category": self.category,
            "tags": list(self.tags),
            "question_altered_count": self.question_altered_count,
        }
